design/one-pager-stacks-versioning.md

```diff
@@ -0,0 +1,248 @@
+ versioning systems linked at the bottom in background research on versioning.
+
+```
+Given a version number MAJOR.MINOR.PATCH, increment the:
+
+MAJOR version when you make incompatible API changes,
+MINOR version when you add functionality in a backwards compatible manner, and
+PATCH version when you make backwards compatible bug fixes.
+Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.
+```
+
+Additional Info: [Semver Reading](https://semver.org/)
+
+Semantic versions for a stack will be managed directly in the channel metadata, and will also be represented in docker
+ tags that point to the same underlying images. To avoid confusion, by convention we should never replace a semantic
+ version tag of software with a different underlying artifact. We should roll forward with a PATCH version. In the event
+ of a security patch we should un-publish the previous stack version when necessary, but ideally after publishing a new
+ higher version first.
+
+### Major Versions
+
+We can assume that major versions like MySQL 5 vs 8 series, might be better served as separate stacks, but the stack
+ system will not place arbitrary restrictions on pushing a major update and assume the author of the stack understands
+ the implications of a major update and can handle it in the stack upgrade logic.
+
+Having update policies should help mitigate major version upgrade problems in the future.
+
+### Enforcement of Version Upgrades
+
+Versions that need to be replaced will be replaced on a roll-forward basis with a new version. We will allow
+ arbitrary version updates that are less than latest to support patch versions to different minor versions of a stack
+ in the event that there are multiple supported minor versions by the software vendor.
+
+Rules for valid version:
+* MAJOR.MINOR.PATCH version in the app.yaml matches the MAJOR.MINOR.PATCH part from the tag ignoring build info in tag.
+* MAJOR.MINOR.PATCH tag must not already be a released stable version else promoting this version makes no sense and
+ would conflict.
+* By convention we can not modify an existing semantic version tag
+
+### Modes for install (Implicit)
+
+Pinned Install: The user will include a fully qualified package name and a specific tag to install. In this mode we will
+ assume an external versioning system is in use and not attempt to check for updates.
+
+Channel Install: If there is a channel set and there is no tag on the package, we will install the latest from the
+ channel. The channel will be checked on an interval and the stacks controller will update the underlying stack based
+ on it's update policy and rules.
+
+### Hosted artifacts
+
+It is assumed that we will have two hosted artifacts at a channel url, i.e.
+
+Channel url:
+https://s3.amazon.com/mybucket/mychannel
+
+Which would contain:
+https://s3.amazon.com/mybucket/mychannel/latest # Latest info
```

Comment:
It feels like these lines are intended to be in a code block, since they are using `#` for comments